### PR TITLE
fix(core): fixes concurrent table create failures

### PR DESCRIPTION
**Pull request by ideoma**
When tables are created by multiple connection concurrently mangled lock files are seen in the logs and table create fails. This was due to usage of shared `Path` member of the `WriterPool`.

Also fixed `CREATE TABLE IF NOT EXISTS` to not throw if in the concurrent environment the table is created between checking for the table existence and trying to lock it on the file system.

### CODE REVIEW

**Review by bluestreak01 — DISMISSED** [review]

File: core/src/test/java/io/questdb/cairo/CreateTableTest.java [hunk 1]
```diff
@@ -283,107 +370,86 @@ public void testCreateTableLikeTableWithMaxUncommittedRowsAndCommitLag() throws
...
+                int threadId = i;
+                threads.add(new Thread(() -> {
+                    try {
+                        barrier.await();
+                        try (
+                                SqlCompiler compiler = new SqlCompiler(engine);
+                                SqlExecutionContextImpl executionContext = new SqlExecutionContextImpl(engine, 1, 1)
+                        ) {
+                            for (int j = 0; j < tableCount; j++) {
+                                compiler.compile("create table if not exists tab" + (threadId * tableCount + j) + " (x int)", executionContext);
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/cairo/CreateTableTest.java:414]: This test shouldn't use IF NOT EXISTS clause, as its name suggests:
```suggestion
                                compiler.compile("create table tab" + (threadId * tableCount + j) + " (x int)", executionContext);
```

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|3|3|100.00%|

**Review by puzpuzpuz — APPROVED** [review]
Nice spot on the race. I wish Java had something similar to ThreadSanitizer.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/pool/WriterPool.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|3|3|100.00%|

**Pull request merged**